### PR TITLE
ColumnLogging.remapFieldKeys()

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
ColumnLogging participating in remapFieldKeys() makes for much more consistent behavior
Also Junit tests

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: api/src/org/labkey/api/data/ColumnLogging.java [hunk 1]
```diff
@@ -82,15 +114,12 @@ public String getOriginalTableName()
...
-    public void setOriginalTableName(String originalTableName)
-    {
-        _originalTableName = originalTableName;
-    }
-
     @Override
     public int compareTo(@NotNull ColumnLogging o)
     {
-        int ret = this.getOriginalTableName().compareToIgnoreCase(o.getOriginalTableName());
+        int ret = this._originalSchemaName.compareTo(o._originalSchemaName);
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/data/ColumnLogging.java]: compareToIgnoreCase() here too?

File: query/src/org/labkey/query/sql/QuerySelectView.java [hunk 2]
```diff
@@ -425,6 +355,86 @@ else if (!shouldLogNameLoggings.isEmpty())
     }
 
 
+    @NotNull
+    private static Set<ColumnInfo> getRequiredDataLoggingColumns(TableInfo table, @NotNull QueryLogging queryLogging, Set<ColumnInfo> allInvolvedColumns, List<ColumnInfo> allColumns, Map<FieldKey, ColumnInfo> columnMap)
+    {
+        // Check allInvolved columns for which need to be logged
+        // Logged columns may also require data logging (e.g. a patientId)
+        // If a data logging column cannot be found, we will only disallow this query (throw an exception)
+        //      if the logged column that requires data logging is in allColumns (which is selected columns plus ones needed for sort/filter)
```
Comment (labkey-jeckels) [inline query/src/org/labkey/query/sql/QuerySelectView.java:364]: I'm not clear on what happens in this case - the description seems to be missing the expected behavior.

**Review by labkey-jeckels — APPROVED** [review]

File: api/src/org/labkey/api/data/ColumnLogging.java (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/data/ColumnLogging.java]: probably better yes.

File: query/src/org/labkey/query/sql/QuerySelectView.java (hunk 2 above)
Comment (labkey-matthewb) [reply query/src/org/labkey/query/sql/QuerySelectView.java:364]: Yes, this is problematic.  None of this code is new.

**Pull request merged**